activation/atxdb.go

```diff
@@ -131,13 +77,13 @@ func init() {
 
 // AwaitAtx returns a channel that will receive notification when the specified atx with id id is received via gossip.
 func (db *DB) AwaitAtx(id types.ATXID) chan struct{} {
-	db.Lock()
-	defer db.Unlock()
-
-	if _, err := db.atxs.Get(getAtxHeaderKey(id)); err == nil {
+	if has, err := atxs.Has(db.sqlDB, id); err == nil && has {
 		return closedChan
 	}
 
+	db.Lock()
```

Comment:
i think this lock should protect whole method. otherwise Await'er may end up waiting on channel that will be never closed